lnwire/onion_error.go

```diff
@@ -1182,6 +1183,26 @@ func (f *InvalidOnionPayload) Encode(w io.Writer, pver uint32) error {
 	return WriteElements(w, f.Offset)
 }
 
+// FailMPPTimeout is returned if the complete amount for a multi part payment was
+// not received within a reasonable time.
+//
+// NOTE: May only be returned by the final node in the path.
+type FailMPPTimeout struct{}
+
+// Code returns the failure unique code.
+//
+// NOTE: Part of the FailureMessage interface.
+func (f *FailMPPTimeout) Code() FailCode {
+	return CodeMPPTimeout
+}
+
+// Returns a human readable string describing the target FailureMessage.
+//
+// NOTE: Implements the error interface.
+func (f *FailMPPTimeout) Error() string {
+	return f.Code().String()
+}
+
```

Comment:
What do we need encode/decode for? I thought they were only necessary for  `EncodeFailureMessage` and `DecodeFailureMessage` if there is additional information to be included (eg a channel update)? Since we optionally check `Serializable` in those methods? 

I've added it to `TestEncodeDecodeCode` (TIL this test exists) and it passes without adding encode/decode, just had to add the code to `makeEmptyOnionError`.

Reply:
Maybe it is worth validating that `lnd` as a sender doesn't do anything strange when it receives this failure. The failure is for example serialized in mission control.

Reply:
@carlaKC you're right my mistake, i see that the other `struct{}` failures also don't have `Encode/Decode` since we don't serialize the messages directly to the wire